blueprints/cloud-operations/network-dashboard/cloud-function/metrics/subnets.py

```diff
@@ -0,0 +1,251 @@
+#
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from . import metrics
+from google.protobuf import field_mask_pb2
+from google.protobuf.json_format import MessageToDict
+import ipaddress
+
+
+def get_all_subnets(config):
+  '''
+    Returns a dictionary with subnet level informations (such as IP utilization)
+    
+      Parameters:
+        config (dict): The dict containing config like clients and limits
+      Returns:
+        subnet_dict (dictionary of String: dictionary): Key is the project_id, value is a nested dictionary with subnet_region/subnet_name as the key.
+  '''
+  subnet_dict = {}
+  subnet_dict = {}
+  read_mask = field_mask_pb2.FieldMask()
+  read_mask.FromJsonString('name,versionedResources')
+
+  response = config["clients"]["asset_client"].search_all_resources(
+      request={
+          "scope": f"organizations/{config['organization']}",
+          "asset_types": ['compute.googleapis.com/Subnetwork'],
+          "read_mask": read_mask,
+      })
+
+  for asset in response:
+    for versioned in asset.versioned_resources:
+      subnet_name = ""
+      network_name = ""
+      project_id = ""
+      ip_cidr_range = ""
+      subnet_region = ""
+
+      for field_name, field_value in versioned.resource.items():
+        if field_name == 'name':
+          subnet_name = field_value
+        if field_name == 'network':
+          # Network self link format:
+          # "https://www.googleapis.com/compute/v1/projects/<PROJECT_ID>/global/networks/<NETWORK_NAME>"
+          project_id = field_value.split('/')[6]
+          network_name = field_value.split('/')[-1]
+        if field_name == 'ipCidrRange':
+          ip_cidr_range = field_value
+        if field_name == 'region':
+          subnet_region = field_value.split('/')[-1]
```

Comment:
```suggestion
        elif field_name == 'network':
          # Network self link format:
          # "https://www.googleapis.com/compute/v1/projects/<PROJECT_ID>/global/networks/<NETWORK_NAME>"
          project_id = field_value.split('/')[6]
          network_name = field_value.split('/')[-1]
        elif field_name == 'ipCidrRange':
          ip_cidr_range = field_value
        elif field_name == 'region':
          subnet_region = field_value.split('/')[-1]
```

this should improve performance at least a bit

Reply:
Thanks!